docs/components/ToastStackView.jsx

```diff
@@ -76,7 +82,10 @@ export default class ToastStackView extends React.PureComponent {
             <ToastStack
               className={cssClass.TOAST_STACK}
               notifications={notifications}
-              setNotifications={ns => this.setState({notifications: ns})}
+              clearNotification={
+                // Internally, `this._clearNotification` performs:
+                // this.setState({notifications: notifications.filter(n => n.id !== id)});
```

Comment:
Added this comment since this is the example code displayed in the Dewey docs

Reply:
Instead of adding a comment, it might be more clear if you inline the function.

Reply:
So inlining the function will actually reproduce the bug that I've been referring to 😅

```tsx
const { notifications } = this.state;
...
clearNotification={(id) => this.setState({
  notifications: notifications.filter(n => n.id !== id)
})
```

Here, I'm creating four notifications in quick succession, each set to expire after a second. Note the (unexpected and repeated) resurfacing of n - 1 notifications without any user interaction :bug:

![bug](https://user-images.githubusercontent.com/12616928/50946392-e8115c80-144d-11e9-8725-a9c1cbe4bd69.gif)

The following tweak fixes the bug while still allowing for inlining in the example code. Key difference: Using `this.state.notifications` instead of the earlier extracted `notifications`.

```tsx
const { notifications } = this.state;
...
clearNotification={(id) => this.setState({
  notifications: this.state.notifications.filter(n => n.id !== id)
})
```

So I could use this. But I'm hesitant for a few reasons.

1. Someone looking at the example code might get confused as to why we sometimes use `this.state.notifications` and other times just `notifications`.
2. I could change everything in the example code to access state fields via `this.state.*` instead of a destructuring `const { ... } = this.state`. But that feels fragile. I could see someone switching to destructuring, not realizing the implications.
3. It's good practice to bind function props like this outside `render` for components that you expect to re-render frequently to avoid repeated function allocations. (More details about this [here](https://medium.freecodecamp.org/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56)!) And I figure the example code should try to demonstrate best practices.

Regarding the bug itself, I would type up an explanation for it, but this comment has already gotten to be pretty lengthy 😅 So let's chat in person tomorrow! tl;dr: `notifications` when extracted from `this.state` can go out of date. Whenever we update the `notifications` state value, `this.state.notifications` becomes a new reference.